contracts/contracts/buyback/BaseBuyback.sol

```diff
@@ -0,0 +1,266 @@
+        ogv = _ogv;
+        cvx = _cvx;
+        cvxLocker = _cvxLocker;
+    }
+
+    /**
+     * @param _uniswapUniversalRouter Address of Uniswap V3 Router
+     * @param _strategistAddr Address of Strategist multi-sig wallet
+     * @param _treasuryManagerAddr Address that receives the treasury's share of OUSD
+     * @param _rewardsSource Address of RewardsSource contract
+     */
+    function initialize(
+        address _uniswapUniversalRouter,
+        address _strategistAddr,
+        address _treasuryManagerAddr,
+        address _rewardsSource
+    ) external onlyGovernor initializer {
+        _setStrategistAddr(_strategistAddr);
+
+        _setUniswapUniversalRouter(_uniswapUniversalRouter);
+        _setRewardsSource(_rewardsSource);
+
+        _setTreasuryManager(_treasuryManagerAddr);
+    }
+
+    /**
+     * @dev Set address of Uniswap Universal Router for performing liquidation
+     * of platform fee tokens. Setting to 0x0 will pause swaps.
+     *
+     * @param _router Address of the Uniswap Universal router
+     */
+    function setUniswapUniversalRouter(address _router) external onlyGovernor {
+        _setUniswapUniversalRouter(_router);
+    }
+
+    function _setUniswapUniversalRouter(address _router) internal {
+        if (universalRouter != address(0)) {
+            // Remove previous router's allowance
+            IERC20(oToken).approve(universalRouter, 0);
+        }
+
+        universalRouter = _router;
+
+        emit UniswapUniversalRouterUpdated(_router);
+    }
+
+    /**
+     * @dev Sets the address that receives the OGV buyback rewards
+     * @param _address Address
+     */
+    function setRewardsSource(address _address) external onlyGovernor {
+        _setRewardsSource(_address);
+    }
+
+    function _setRewardsSource(address _address) internal {
+        require(_address != address(0), "Address not set");
+        rewardsSource = _address;
+        emit RewardsSourceUpdated(_address);
+    }
+
+    /**
+     * @dev Sets the address that can receive and manage the funds for Treasury
+     * @param _address Address
+     */
+    function setTreasuryManager(address _address) external onlyGovernor {
+        _setTreasuryManager(_address);
+    }
+
+    function _setTreasuryManager(address _address) internal {
+        require(_address != address(0), "Address not set");
+        treasuryManager = _address;
+        emit TreasuryManagerUpdated(_address);
+    }
+
+    /**
+     * @dev Swaps half of `oTokenAmount` to OGV
+     *      and the rest to CVX and finally lock up CVX
+     * @param oTokenAmount Amount of OUSD/OETH to swap
+     * @param minOGV Minimum OGV to receive for oTokenAmount/2
+     * @param minCVX Minimum CVX to receive for oTokenAmount/2
+     */
+    function swap(
+        uint256 oTokenAmount,
+        uint256 minOGV,
+        uint256 minCVX
+    ) external onlyGovernorOrStrategist nonReentrant {
+        _swap(oTokenAmount, minOGV, minCVX);
+    }
+
+    function _swap(
```

Comment:
Do we need a separate internal function here for swap? Can this code just be in the external swap() function?

Reply:
Yep, Have removed the internal function